src/library_pipefs.js

```diff
@@ -0,0 +1,208 @@
+mergeInto(LibraryManager.library, {
+  $PIPEFS__postset: '__ATINIT__.push(function() { PIPEFS.root = FS.mount(PIPEFS, {}, null); });',
+  $PIPEFS__deps: ['$FS'],
+  $PIPEFS: {
+    BUCKET_BUFFER_SIZE: 1024 * 8, // 8KiB Buffer
+    mount: function (mount) {
+      return FS.createNode(null, '/', {{{ cDefine('S_IFDIR') }}} | 511 /* 0777 */, 0);
```

Comment:
`SOCKFS` implements BSD sockets support via a WebSockets backend, it is currently supported. Reading closer, this is not mounting `PIPEFS` to the root of the filesystem, this is calling `FS.createNode` (and not `FS.mount`) to create the filesystem root directory. This should be redundant, the root directory should exist, and this actually would create multiple versions of that node. Does `PIPEFS.mount` get called from anywhere? It can be removed if not.